lib/core.ml

```diff
@@ -3,12 +3,14 @@
 open Packet
 open Ciphersuite
 
-let (<+>) = Cstruct.append
-
 let ( let* ) = Result.bind
 
 let guard p e = if p then Ok () else Error e
 
+let split_str ?(start = 0) str off =
+  String.sub str start off,
+  String.sub str (start + off) (String.length str - off - start)
```

Comment:
it's likely I introduced this function, but from our learnings in MirageVPN, I guess we should re-visit the allocation and ownership stuff in this PR. The Strong.sub is super-expensive, and we should try to allocate the least possible.

This doesn't need to be in this PR, but before a release. We should as well benchmark the amount of handshakes and throughput before and after this change (with varying block sizes in send()). Not sure whether we have a suitable executable for this right now. Best to compare to OpenSSL on the same machine.